aws/stage/es/main.tf

```
@@ -29,7 +29,40 @@ resource "ec_deployment" "es_cluster" {
     ec_deployment_traffic_filter.aws_vpce.id
   ]
 
-  elasticsearch {}
+  elasticsearch {
+    autoscale = "true"
+
+    topology {
+      id = "cold"
+    }
+
+    topology {
+      id = "frozen"
+    }
+
+    topology {
+      id   = "hot_content"
+      size = "8g"
+
+      autoscaling {
+        max_size = "${var.elastic_memory_max}g"
+      }
+    }
+    topology {
+      id = "ml"
+    }
+
+    topology {
+      id = "warm"
+    }
+  }
+
+  lifecycle {
+    ignore_changes = [
+      elasticsearch[0].topology[2].size
+    ]
+  }
```

we can keep the initial size to 1g to keep the cost down (until it is used)
i think the other topology blocks are not required because there are not in the plan and the max_size was zero in the UI

```suggestion
  elasticsearch {
    autoscale = "true"
  
    topology {
      id   = "hot_content"
      size = "1g"

      autoscaling {
        max_size = "${var.elastic_memory_max}g"
      }
    }
  }

  lifecycle {
    ignore_changes = [
      elasticsearch[0].topology[0].size
    ]
  }
```

---

Ok I'll lowered it down. The other topologies are needed. As soon as you enable autoscaling whenever you try to apply terraform it will tell you that they are missing, because autoscaling enables them.

---

ok then we keep them 👍